cli/tests/e2e/test_ci.py

```diff
@@ -1,3 +1,13 @@
+##############################################################################
+# Prelude
+##############################################################################
+# Testing 'semgrep ci' "end-to-end".
+#
+# TODO: actually most of the tests in this file rely on use_click_runner=True
+# because of some mocking and monkeypatching. Thus, this is this not
+# a real e2e test because cli/bin/semgrep is not invoked.
+# Try to use environment variables instead of Python monkey patching
+# so that those tests can also pass with osemgrep.
```

Comment:
This is a little sad to hear. What percentage of tests would you estimate are truly running end to end? It will be interesting to see how much work there is to merely convert the tests to being truly end to end. As I understand it, this would be a prerequisite before porting the code that would need to be tested by these tests anyway. Is that correct?

Reply:
there's 40 tests_xxx.py in cli/tests/e2e/, so it's not too bad that currently just one file (well test_fixtest.py and test_ignores.py have a few use_click_runner=True but we should be able to fix them easily) requires this flag.
Anyway, most of our effort in osemgrep is on porting 'semgrep scan' and we are delaying porting 'semgrep ci' which is tested by test_ci.py, so we're fine I think.